cloud/cluster/cluster.go

```diff
@@ -0,0 +1,65 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
+}
+
+type NodeUpdateType int
+
+const (
+	NodeAdded NodeUpdateType = iota
+	NodeRemoved
+)
+
+type NodeUpdate struct {
+	UpdateType NodeUpdateType
+	Id         NodeId
+	Node       Node // Only set for adds
+}
+
+func (u *NodeUpdate) String() string {
+	return fmt.Sprintf("%v %v %v", u.UpdateType, u.Id, u.Node)
+}
+
+type Cluster interface {
+	Members() ([]Node, error)
+	Subscribe() (Subscription, error)
+	Close() error
+}
+
+type Subscription struct {
+	InitialMembers []Node
+	Updates        chan []NodeUpdate
+	Closer         io.Closer
+}
+
+// NodeWrapper lets users wrap a Node.
+// E.g., if service discovery hands you an address, you can wrap it with a Node that also contains
+// the Thrift client to talk to that address
+type NodeWrapper interface {
```

Comment:
I don't totally understand what the NodeWrapper buys us over the Interface?  Why not just have factory methods that return an implementation of this interface

The idea behind the interface is that it allows any implementation to be called by the Scheduler.  The current code now couples the scheduler to NodeWorker etc...

The interface also defined types of errors that the node could return.  The wrapper defines no common errors?  The scheduler will eventually need some insight on the types of possible errors that could be returned when implementing dead letter queues etc...